src/sys/wait.rs

```diff
@@ -184,6 +191,12 @@ fn decode(pid : pid_t, status: i32) -> WaitStatus {
     } else if status::signaled(status) {
         WaitStatus::Signaled(pid, status::term_signal(status), status::dumped_core(status))
     } else if status::stopped(status) {
+        #[cfg(any(target_os = "linux", target_os = "android"))] {
```

Comment:
I didn't know `cfg` on blocks was possible, even if nightly only.

To do this on stable, I think your best approach is going to be adding an inner function like this:

```rust
cfg_if! {
    if #[cfg(any(target_os = "linux", target_os = "android"))] {
        fn decode_stopped(status) -> WaitStatus {
            // Linux code
        }
    } else {
        fn decode_stopped(status) -> WaitStatus {
            // Other platform code
        }

        decode_stopped(status)
    }
}
```